core/src/main/java/com/datastrato/gravitino/meta/ManagedUser.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.meta;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.Field;
+import com.datastrato.gravitino.HasIdentifier;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.authorization.User;
+import com.google.common.collect.Maps;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import lombok.ToString;
+
+/** A class representing a user metadata entity in Gravitino. */
+@ToString
+public class ManagedUser implements User, Entity, Auditable, HasIdentifier {
```

Comment:
Can we just directly use `UserEntity`, instead of using this `ManagedUser`, actually Gravitino doesn't manage users.